projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -599,7 +607,37 @@ public function get_item_permissions_check( $request ) { //phpcs:ignore Variable
 	}
 
 	/**
-	 * Get status for all supported integrations.
+	 * Core logic for a single integration
+	 *
+	 * @param string $slug Integration slug.
+	 * @return array Integration status data.
+	 */
+	private function get_integration( $slug ) {
+		$config       = $this->get_supported_integrations()[ $slug ];
+		$status       = $config['type'] === 'plugin'
+			? $this->get_plugin_status( $slug )
+			: $this->get_service_status( $slug );
+		$status['id'] = $slug;
+		return $status;
+	}
```

Comment:
New method so both the /integrations endpoint and /integrations/slug will be pulling data from the same method for data consistency.